src/tiled/wangfiller.cpp

```diff
@@ -47,26 +50,68 @@ void WangFiller::setWangSet(WangSet *wangSet)
     mWangSet = wangSet;
 }
 
+static QPoint *getSurroundingPoints(QPoint point, StaggeredRenderer *staggeredRenderer, Map::StaggerAxis staggerAxis)
+{
+    QPoint *adjacentPoints = (QPoint*)malloc(sizeof(QPoint) * 8);
```

Comment:
There's still the problem with `malloc` / `delete` mismatch. Actually, I suggested using a static array but I think it's not going to work because the function is called multiple times. But you could allocate these arrays on the stack and pass them into the function:
```c++
static void getSurroundingPoints(QPoint point, StaggeredRenderer *staggeredRenderer, Map::StaggerAxis staggerAxis, QPoint *points)

QPoint adjacentPoints[8];
getSurroundingPoints(point, staggeredRenderer, staggerAxis, adjacentPoints);
```